lib/mongoose_push_web/controllers/api_v1/notification_controller.ex

```diff
@@ -51,9 +51,13 @@ defmodule MongoosePushWeb.APIv1.NotificationController do
   end
 
   def send(conn = %{body_params: params}, %{device_id: device_id}) do
-    result = MongoosePush.Application.backend_module().push(device_id, params)
+    request = MongoosePushWeb.APIv1.RequestDecoder.decode(params)
+    result = MongoosePush.Application.backend_module().push(device_id, request)
     {status, payload} = MongoosePush.API.V1.ResponseEncoder.to_status(result)
 
+    # TODO remove the line below after the reimplementation of CastAndValidate plug
+    conn = update_in(conn.body_params, &Map.from_struct(&1))
```

Comment:
I don't quite understand why we do that here

Reply:
We are omitting dialyzer issues I mentioned in PR summary. More elaborative comment in `0b338fe9d2f3` description.